lib/queryContextCache.js

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+ */
+const Logger = require('./logger');
+const crypto = require('crypto');
+/**
+     * Constructor.
+     *
+     * @param id database id
+     * @param timestamp Server time when this entry read
+     * @param priority Priority of this entry w.r.t other ids
+     * @param context Opaque query context, used by query processor in the server.
+     */
+
+/** Query context information. */
+function QueryContextElement (id,timestamp,priority,context) {
+  this.id = id;
+  this.timestamp = timestamp;
+  this.priority = priority;
+  this.context = context;
+}
+
+QueryContextElement.prototype.getId = function () {
```

Comment:
do we need those getters, setters here?

Reply:
I think it's quite usual to access fields directly in javascript - especially that it's simple dto

Reply:
Okay, I will use the direct access then.

Reply:
Removed all getters and setters for Query Context Elements, and only used the direct access.